initiator.go

```diff
@@ -366,27 +374,28 @@ func iscsiBoolStr(pred bool) string {
 func (s *IscsiTargetSession) SetParams() error {
 	params := []struct {
 		p IscsiParam
+        s string
 		v string
 	}{
-		{ISCSI_PARAM_TARGET_NAME, s.opts.Volume},
-		{ISCSI_PARAM_INITIATOR_NAME, "iscsi_startup.go"},
-		{ISCSI_PARAM_MAX_RECV_DLENGTH, fmt.Sprintf("%d", s.opts.MaxRecvDLength)},
-		{ISCSI_PARAM_MAX_XMIT_DLENGTH, fmt.Sprintf("%d", s.opts.MaxXmitDLength)},
-		{ISCSI_PARAM_FIRST_BURST, fmt.Sprintf("%d", s.opts.FirstBurstLength)},
-		{ISCSI_PARAM_MAX_BURST, fmt.Sprintf("%d", s.opts.MaxBurstLength)},
-		{ISCSI_PARAM_PDU_INORDER_EN, netlinkBoolStr(s.opts.DataPDUInOrder)},
-		{ISCSI_PARAM_DATASEQ_INORDER_EN, netlinkBoolStr(s.opts.DataSequenceInOrder)},
-		{ISCSI_PARAM_INITIAL_R2T_EN, netlinkBoolStr(s.opts.InitialR2T)},
-		{ISCSI_PARAM_IMM_DATA_EN, netlinkBoolStr(s.opts.ImmediateData)},
-		{ISCSI_PARAM_EXP_STATSN, fmt.Sprintf("%d", s.expStatSN)},
-		{ISCSI_PARAM_HDRDGST_EN, netlinkBoolStr(s.opts.HeaderDigest == "CRC32C")},
-		{ISCSI_PARAM_DATADGST_EN, netlinkBoolStr(s.opts.DataDigest == "CRC32C")},
-		{ISCSI_PARAM_PING_TMO, fmt.Sprintf("%d", s.opts.PingTimeout)},
-		{ISCSI_PARAM_RECV_TMO, fmt.Sprintf("%d", s.opts.RecvTimeout)},
+		{ISCSI_PARAM_TARGET_NAME, "Target Name", s.opts.Volume},
```

Comment:
I would have made IscsiParam have a String() function to return this

Reply:
Wouldnt that mean i would need a switch  case for every ISCSI_PARAM_XXX. I actually included this code just for log readability. I can take it out if we dont need it. 

From what i am understanding, this would mean
func (p IscsiParam) String() {
switch(p)
case ISCSI_PARAM_TARGET_NAME:
   return "Target Name"
......
}
Is that what you had in mind ?
}

Reply:
that or a map[IscsiParam]string that you use in the String() method.